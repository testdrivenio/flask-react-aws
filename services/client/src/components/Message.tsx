import React from "react";
import {
  Alert,
  AlertIcon,
  AlertTitle,
  CloseButton,
  Box,
  Container,
} from "@chakra-ui/react";

interface MessageProps {
  messageType: "info" | "warning" | "success" | "error";
  messageText: string;
  onClose?: () => void;
}

const Message: React.FC<MessageProps> = ({
  messageType,
  messageText,
  onClose,
}) => {
  return (
    <Container maxW="container.2xl" px={4}>
      <Alert
        status={messageType}
        variant="solid"
        mt={4}
        borderRadius="md"
        display="flex"
        alignItems="center"
      >
        <AlertIcon />
        <Box flex="1">
          <AlertTitle>{messageText}</AlertTitle>
        </Box>
        {onClose && (
          <CloseButton
            onClick={onClose}
            ml={2}
            size="sm"
            color="white"
            _hover={{ bg: "whiteAlpha.300" }}
          />
        )}
      </Alert>
    </Container>
  );
};

export default Message;
