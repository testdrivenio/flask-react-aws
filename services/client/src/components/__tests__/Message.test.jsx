import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Message from '../Message';

afterEach(cleanup);

describe('when "messageType" is "success"', () => {
  const props = {
    messageType: 'success',
    messageText: 'Hello, World!',
    removeMessage: () => true,
  }

  it('renders the default props', async() => {
    const { getByText, getByTestId } = render(<Message {...props} />);
    expect(getByTestId('message').innerHTML).toContain('is-success');
    expect(getByText('Hello, World!')).toHaveClass('message-text');
  });

  it("renders", () => {
    const { asFragment } = render(<Message {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('when "messageType" is "danger"', () => {
  const props = {
    messageType: 'danger',
    messageText: 'Hello, World!',
    removeMessage: () => true,
  }

  it('renders the default props', () => {
    const { getByText, getByTestId } = render(<Message {...props} />);
    expect(getByTestId('message').innerHTML).toContain('is-danger');
    expect(getByText('Hello, World!')).toHaveClass('message-text');
  });

  it("renders", () => {
    const { asFragment } = render(<Message {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
