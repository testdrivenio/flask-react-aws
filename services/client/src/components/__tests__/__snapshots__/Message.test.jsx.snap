// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when "messageType" is "danger" renders 1`] = `
<DocumentFragment>
  <section
    data-testid="message"
  >
    <div
      class="notification is-danger"
    >
      <button
        class="delete"
      />
      <span
        class="message-text"
      >
        Hello, World!
      </span>
    </div>
  </section>
</DocumentFragment>
`;

exports[`when "messageType" is "success" renders 1`] = `
<DocumentFragment>
  <section
    data-testid="message"
  >
    <div
      class="notification is-success"
    >
      <button
        class="delete"
      />
      <span
        class="message-text"
      >
        Hello, World!
      </span>
    </div>
  </section>
</DocumentFragment>
`;
